src/test/java/seedu/address/model/room/RoomTest.java

```diff
@@ -0,0 +1,100 @@
+package seedu.address.model.room;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_CAPACITY_DOUBLE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_ROOM_NUMBER_074;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_HANDICAP;
+import static seedu.address.testutil.TypicalRooms.DOUBLE_002;
+import static seedu.address.testutil.TypicalRooms.SINGLE_001;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import seedu.address.testutil.RoomBuilder;
+import seedu.address.testutil.TypicalBookings;
+
+public class RoomTest {
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void asObservableList_modifyList_throwsUnsupportedOperationException() {
+        Room room = new RoomBuilder().build();
+        thrown.expect(UnsupportedOperationException.class);
+        room.getTags().remove(0);
+    }
+
+    @Test
+    public void isSameRoom() {
+        // same object -> returns true
+        assertTrue(SINGLE_001.isSameRoom(SINGLE_001));
+
+        // null -> returns false
+        assertFalse(SINGLE_001.isSameRoom(null));
+
+        // different room number -> returns false
+        Room editedSingle001 = new RoomBuilder(SINGLE_001).withRoomNumber("002").build();
+        assertFalse(SINGLE_001.isSameRoom(editedSingle001));
+
+        // same room number, different capacity -> returns true
```

Comment:
This is a good defensive test case.

Technically, the capacity of the room should not be able to be edited. Is this ensured elsewhere?

Reply:
Yes. 
No methods that can edit a room capacity. 
Rooms can only be instantiated as `SingleRoom`, `DoubleRoom`, or `SuiteRoom`, since `Room` is an abstract class. Any changes that result in an invalid Capacity in the text file will result in a IllegalValueException raised.